core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -373,7 +373,7 @@ private void applyOutstandingWalTransactions(
 
                             default:
                                 // Always set full path when using thread static path
-                                operationExecutor.setNowAndFixClock(commitTimestamp);
+                                operationExecutor.setNowAndFixClock(commitTimestamp, writer.getTimestampType());
```

Comment:
this is wild


I think the logic here is to "freeze" `now()` values. Moving clock type dynamically feels like a bad idea. We just need to freeze time - for example we can introduce frozen time state for both micro- and nano-now() (if we have it). We must also isolate "now()" clock from other clocks. We are returning random clock as `getMicrosecondClock()` - it is a spaghetti at best and disaster at worst.

Reply:
fixed.
Introduced micro and nanos precision for `now`.